torch_xla/csrc/tensor_methods.cpp

```diff
@@ -1552,12 +1544,25 @@ XLATensor XLATensor::lerp(const XLATensor& input, const XLATensor& end,
 }
 
 XLATensor XLATensor::log(const XLATensor& input) {
-  return input.CreateFrom(ir::ops::Log(input.GetIrValue()));
+  // Here we explictly pass c10::nullopt as logical_element_type because
+  // otherwise result will inherit the input's logical_element_type. In the
+  // case of log(int) -> float, we want to derive the dtype from IR value
+  // instead of input's logical_element_type.
+  return input.CreateFrom(
+      ir::ops::Log(GetFloatingIrValue(input, at::ScalarType::Float)),
```

Comment:
`GetFloatingIrValue` only apply when the input is not a floating type, so `int8`, `int16`, `int32` and `int64` (and uint version). Ideally we should use `float64` for `int64` but the likelihood of people using `log(int64)` is low I was being a bit lazy. (If user do `log(int)` it will fail today and we never have people complain about it).

Reply:
My question is about the `complex` data type. How does this handle complex numbers?

Reply:
all complex type we supported are float https://github.com/pytorch/xla/blob/9ddc27d88c1e974a5fe3261acbf53101d43238a8/torch_xla/csrc/tensor_util.cpp#L115

Reply:
Oh ok, I was wrong.. I guess I can explicitly check for int type for now